src/main/kotlin/ar/edu/unahur/obj2/servidorWeb/ServidorWeb.kt

```diff
@@ -10,5 +8,81 @@ enum class CodigoHttp(val codigo: Int) {
   NOT_FOUND(404),
 }
 
-class Pedido(val ip: String, val url: String, val fechaHora: LocalDateTime)
-class Respuesta(val codigo: CodigoHttp, val body: String, val tiempo: Int, val pedido: Pedido)
+class ServidorWeb() {
+  val modulos = mutableListOf<Modulo>()
+  val analizadores = mutableListOf<Analizador>()
+
+  fun agregarModulo(modulo : Modulo){
+    modulos.add(modulo)
+  }
+
+  fun quitarModulo(modulo: Modulo){
+    modulos.remove(modulo)
+  }
+
+  fun agregarAnalizador(analizador : Analizador){
+    analizadores.add(analizador)
+  }
+
+  fun quitarAnalizador(analizador:Analizador){
+    analizadores.remove(analizador)
+  }
+
+  fun listaDeModulosNoEstaVacia() = modulos.isNotEmpty()
+
+  fun listaDeAnalizadoresNoEstaVacia() = analizadores.isNotEmpty()
+
+  fun verificarProtocolo(pedido : Pedido) : Respuesta {
+    return if (pedido.protocolo() != "http")
+      Respuesta(CodigoHttp.NOT_IMPLEMENTED, "", 10, pedido)
+    else
+      Respuesta(CodigoHttp.OK, "hola, esta es una respuesta del servidor", 12, pedido)
+  }
+
+  fun buscarModuloQuePuedaTrabajarCon(pedido : Pedido) : Modulo{
+    return modulos.find{it.puedeTrabajar(pedido.extension())}!!
+  }
+
+  fun existeModuloQuePuedaTrabajarCon(pedido: Pedido) : Boolean{
+    return modulos.any{it.puedeTrabajar(pedido.extension())}
+  }
+
+  fun respuestaDelModulo(pedido : Pedido) : Respuesta{
+    return if(!this.existeModuloQuePuedaTrabajarCon(pedido)){
+      Respuesta(CodigoHttp.NOT_FOUND,"",10,pedido)
+    }else{
+      this.buscarModuloQuePuedaTrabajarCon(pedido).respuestaDelModulo(pedido)
+    }
+  }
+
+  fun mandarAAnalizar(respuestaFinal: Respuesta, modulo : Modulo) =
+    analizadores.forEach{it.analizar(respuestaFinal,modulo)}
+
+
+  fun atenderPedido(pedido: Pedido) :Respuesta{
+    lateinit var respuestaObtenida : Respuesta
+    lateinit var moduloSeleccionado : Modulo
+
+    respuestaObtenida = this.verificarProtocolo(pedido)
+
+    if (this.listaDeModulosNoEstaVacia()){
+      if(this.existeModuloQuePuedaTrabajarCon(pedido)) {
+        moduloSeleccionado = this.buscarModuloQuePuedaTrabajarCon(pedido)
+      }
+      respuestaObtenida = this.respuestaDelModulo(pedido)
+    }
+
+    if(this.listaDeAnalizadoresNoEstaVacia() && this.listaDeModulosNoEstaVacia()){
+      this.mandarAAnalizar(respuestaObtenida, moduloSeleccionado)
```

Comment:
Si `this.listaDeAnalizadoresNoEstaVacia()` && `this.listaDeModulosNoEstaVacia()` && `!this.existeModuloQuePuedaTrabajarCon(pedido)` esto rompe, ¿no?